source/common/tcp_proxy/upstream.h

```diff
@@ -77,7 +77,7 @@ class HttpConnPool : public GenericConnPool, public Http::ConnectionPool::Callba
     virtual void onFailure() {
       ASSERT(conn_pool_ != nullptr);
       conn_pool_->callbacks_->onGenericPoolFailure(
-          ConnectionPool::PoolFailureReason::RemoteConnectionFailure, host_);
+          ConnectionPool::PoolFailureReason::RemoteConnectionFailure, "", host_);
```

Comment:
Empty reason string view is ok here?

Reply:
I did not want to change the behavior here.

Reply:
Sorry. But I still think this behavior change is what we want. The whole PR is try to log the `UPSTREAM_TRANSPORT_FAILURE_REASON`, right? Then a more specific `UPSTREAM_TRANSPORT_FAILURE_REASON` would be helpful.

Reply:
so my concern with having the empty string here, is that I don't know if onFailure will ever be called for issues other than failure of the CONNECT handshake.  If you want to leave an empty string here, how about renaming the function onConnectHandshakeFailure or something and then add a comment that we're doing an empty string because it's not a transport failure?

though honestly I'd be inclined to do that rename, have a string connect_handshake_failed, and leave things failure_reason rather than transport_failure_reason - it's helpful to know why upstream failed to connect in all cases IMO, not just for TCP/TLS failures.
your call though :-)